### PR TITLE
[SPARK-41717][CONNECT] Deduplicate print and _repr_html_ at LogicalPlan

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR deduplicates `print` and `_repr_html_` logic by having a comment logic to handle them by looking up the signature and type hints (similar with what we're doing at Spark SQL's query string, e.g., `TreeNode.stringArgs`).

### Why are the changes needed?

To make it easier to maintain.

### Does this PR introduce _any_ user-facing change?

Virtually no.

### How was this patch tested?

I tested almost all cases, and complex cases too manually. In addition, I added a unittest too.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
cc @amaliujia @zhengruifeng @grundprinzip FYI

**Review by zhengruifeng — APPROVED** [review]
thanks for doing this!

**Comment by HyukjinKwon** [comment]
Related tests passed.

Merged to master.

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -95,14 +96,68 @@ def to_proto(self, session: "SparkConnectClient", debug: bool = False) -> proto.
 
         return plan
 
-    # TODO(SPARK-41717): Implement the command logic for print and _repr_html_
+    def _parameters_to_print(self, parameters: Mapping[str, Any]) -> Mapping[str, Any]:
+        params = {}
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/plan.py:100]: The logic of this method is complicated enough to deserve a docstring.

File: python/pyspark/sql/connect/plan.py [hunk 2]
```diff
@@ -207,7 +240,7 @@ def print(self, indent: int = 0) -> str:
     def _repr_html_(self) -> str:
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/plan.py:240]: Why is this method not removed?

File: python/pyspark/sql/connect/plan.py [hunk 3]
```diff
@@ -265,13 +277,13 @@ class Project(LogicalPlan):
 
     def __init__(self, child: Optional["LogicalPlan"], *columns: "ColumnOrName") -> None:
         super().__init__(child)
-        self._raw_columns = list(columns)
```
Comment (grundprinzip) [inline python/pyspark/sql/connect/plan.py:268]: Why this change?

**Comment by grundprinzip** [comment]
Thanks for doing it. Please prepare a follow-up PR that adds documentation to the parameter generating function because that is not "trivial" and thus deserves more explanation.

File: python/pyspark/sql/connect/plan.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/plan.py:100]: Sure, let me add some docs.

File: python/pyspark/sql/connect/plan.py (hunk 2 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/plan.py:240]: This wasn't removed because `LocalRelation` contains `data` that's PyArrow table, and this isn't printed out pretty together with our format out of the box. So it was manually overridden here.

File: python/pyspark/sql/connect/plan.py (hunk 3 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/plan.py:268]: Because this one was only the exception that does not follow the variable naming style in this file. All are either `name` or `_name`, and the automatic printing logic relies on this naming.

File: python/pyspark/sql/connect/plan.py (hunk 1 above)
Comment (HyukjinKwon) [reply python/pyspark/sql/connect/plan.py:100]: Here https://github.com/apache/spark/pull/39237
